src/res_printer.ml

```diff
@@ -2494,6 +2494,7 @@ and printExpression (e : Parsetree.expression) cmtTbl =
   | Parsetree.Pexp_constant c -> printConstant c
   | Pexp_construct _ when ParsetreeViewer.hasJsxAttribute e.pexp_attributes ->
     printJsxFragment e cmtTbl
+  | Pexp_construct ({txt = Longident.Lident "()"; loc}, _) when loc.loc_ghost -> Doc.nil
```

Comment:
Maybe add a comment here. It'll be unclear where this comes from.
Also can't we _just_ make it so that `{}` is another sugar for `()` in these circumstances? And on the printer side print `()` as `{}` in the right circumstances too?
I'm guessing that code would be slightly more verbose on the printer side but maybe less complexity/knowledge needed?